SubscribeIT/ngDesk
# 33 exception for signature nodes

#### Reference to issue (Require

#### Checklist (Required)


#### Description (Required)

Bugs in signature-nodes

#### Manual Test Preformed (Required)
List the manual tests that were preformed and the steps to preform them  
**There must be multiple test cases provided, each test case should include the exact and detailed steps to replicate the case**
**I have included 3, test cases here but there will be many time when more cases are required**


##### Test Case 1
**Name:Create Custom module with all data type fields**

- log in ngdesk
- go to module sidebar
- create new module
- create fields with different data type
- add fields in edit,create and list layouts
- add newly custom module in sidebar
- able to create edit entries of custom module without any error
- chronometer fields working without any error



##### Test Case 2
**Name: Create PDF with all field **
- go to module sidebar -> select custom module
- go to pdf 
- Create new pdf with all field 
- save it.


##### Test Case 3
**Name:Craete signature document workflow**

- module sidebar -> select custom module
- go to workflow
- create signature workflow without any condition
- go to customize  and in template select newly generated pdf and give other details.
- save it


##### Test Case 4
**Name:Craete new entry of custom module and check signature doc**

- go to custom module sidebar
- create new entries with all details
- save it
- go to signature document collection in DB,copy id
- go to https://dev1.ngdesk.com/document-viewer/signature_doc_id
- go to sign to document option
- put signature
- able save it 
- sign key and datesigned field updated in DB


Test Case 5

Name: Create a field of data type chronometer in Tickets module


1. Login to subdomain.
2. Goto -> Modules -> Tickets -> Fields -> New.
3. Create a new field of chronometer data type and hit save.
4. Goto -> Modules -> Tickets -> Fields -> Layout.
5. Add the created field (of chronometer data type) to the web edit layout and hit save.
6. Goto -> Tickets -> New.
7. Create a new Ticket and hit save.
8. Open the created ticket and observe.
9. Observe no error.



#### List of General Components affected (Required)


#### Types of changes (Required)

####Screenshots
![Screenshot from 2021-08-31 15-37-51](https://user-images.githubusercontent.com/89505098/131484320-e7863c7a-503c-45cf-a217-42f1b1470649.png)
![Screenshot from 2021-08-31 15-37-33](https://user-images.githubusercontent.com/89505098/131484331-cb9ffca0-a8ba-4478-a4c2-41a6e0a4947f.png)
![Screenshot from 2021-08-31 15-36-48](https://user-images.githubusercontent.com/89505098/131484334-b6d8dfdb-20c6-48e9-b0c5-d560525a6246.png)
![Screenshot from 2021-08-31 15-35-44](https://user-images.githubusercontent.com/89505098/131484341-796d31a2-ba46-4359-bec8-9c61caba0a5c.png)
![Screenshot from 2021-08-31 15-35-37](https://user-images.githubusercontent.com/89505098/131484346-675e294c-2024-4601-b994-07d6f95b4c3e.png)
![Screenshot from 2021-08-31 15-42-18](https://user-images.githubusercontent.com/89505098/131485028-bfce31d1-74c4-40d9-8dd3-1d0a813c559f.png)
![Screenshot from 2021-09-09 11-36-24](https://user-images.githubusercontent.com/89562536/132632813-5895686c-a84c-4359-b674-51629907e6c8.png)
![Screenshot from 2021-09-09 11-36-46](https://user-images.githubusercontent.com/89562536/132632845-7127719f-b029-44e1-a986-03e23e7e6e54.png)
